make/xml.cmake

```diff
@@ -37,3 +37,99 @@ function(XML_LINT)
 
 endfunction(XML_LINT)
 
+function(XML_SORT)
+  # ~~~
+  # XML_SORT(
+  # NAME
+  # FILE
+  # OUTPUT
+  # )
+  set(oneValueArgs NAME FILE OUTPUT)
+  cmake_parse_arguments(
+    XML_SORT
+    ""
+    "${oneValueArgs}"
+    ""
+    ${ARGN}
+    )
+
+  set(XML_SORT_XSL ${symbiflow-arch-defs_SOURCE_DIR}/common/xml/xmlsort.xsl)
+  set(
+    XML_SORT_INPUT ${CMAKE_CURRENT_BINARY_DIR}/${XML_SORT_FILE}
+  )
+  get_file_target(XML_SORT_INPUT_TARGET ${XML_SORT_FILE})
+  get_target_property(INCLUDE_FILES ${XML_SORT_INPUT_TARGET} INCLUDE_FILES)
+  set(DEPS "")
+  foreach(SRC ${INCLUDE_FILES})
+    append_file_dependency(DEPS ${SRC})
+  endforeach()
+  set(XML_SORT_OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/${XML_SORT_OUTPUT})
+
+  get_target_property_required(XSLTPROC env XSLTPROC)
+  get_target_property(XSLTPROC_TARGET env XSLTPROC_TARGET)
+
+  add_custom_command(
+    OUTPUT ${XML_SORT_OUTPUT}
+    DEPENDS
+      ${XML_SORT_XSL}
+      ${XML_SORT_INPUT}
+      ${XML_SORT_INPUT_TARGET}
+      ${DEPS}
+      ${XSLTPROC} ${XSLTPROC_TARGET}
+    COMMAND
+      ${CMAKE_COMMAND} -E make_directory
+      ${CMAKE_CURRENT_BINARY_DIR}/${OUT_DEVICE_DIR}
+    COMMAND
+      ${XSLTPROC}
+      --nomkdir
+      --nonet
+      --xinclude
+      --output ${XML_SORT_OUTPUT} ${XML_SORT_XSL} ${XML_SORT_INPUT}
+  )
+  add_custom_target(
```

Comment:
Need to call add_file_target(FILE ${XML_SORT_OUTPUT} GENERATED)

Reply:
Done! - Should this be done before or after the `add_custom_command`?

Reply:
I recommend after, but I don't think it actually matters.

Reply:
Done.